### ISSUE TITLE
Inspect and remove unnecessary no-arc flags.

### ISSUE THREAD

**Issue by lolgear** [git-up/GitUp#825]
I noticed that many files have no-arc flag.

I inspected these files and noticed that in some places they may be necessary, but in most places ARC can take of everything.

For example, most common pattern is property and _ivar.
But if you want to not to use retain/release in property assignments with "strong" attribute, you can use "unsafe_unretained" which explicitly removes "-retain" message in assignment.

I suggest to inspect files and remove these flags ( as many as we can ) and make everything under ARC ( especially, on modern devices it may have no sense to run for these kind of optimizations ).

**Comment by lucasderraugh** [comment]
Yup, it's mostly left to tricky cases. Any conversion would be good.

**Comment by lolgear** [comment]
@lucasderraugh 
I removed everything, but, of course, it crashes.
You can check whole branch https://github.com/lolgear/GitUp/tree/refactoring/mrc_removal
If you have any suggestions, then, feel free to merge/change.

**Comment by lolgear** [comment]
I encountered with several typical "transforms" during branch preparation:

1. Bridge casting.
2. Self release in initializers.
3. Retain/Release calls at the beginning/ at the end of variables scopes.
4. CoreFoundation types and their handling.
5. Initializers with retaining values for ivars.
6. Implicit ivars generation.
7. Gotos and order of variables initialization.
8. Calling extrenal ( libgit2 ) libraries.

Some of them are very easy to fix, others are a bit tricky.

**Comment by lolgear** [comment]
I also have several suggestions regarding current GCPrivate/GIPrivate.

### GCPrivate/GIPrivate separation into small pieces.
These two headers can be separated in many small GCEntity+Private/GIEntity+Private headers.
This change simplifies searching for interfaces/extensions/categories.

Also one curious thing that I don't understand:

### Xcode and autogenerated ivars for readonly properties?
How does Xcode generates ivars with readonly modifier in property?
For example, GINode has commit property which has readonly modifier.
So, Xcode doesn't know how to create ivar even in MRC environment, right?

### CFArray vs NSArray
And another thing is about retain/release in CFArray/NSArray ( and their mutable counterparts ).
This pattern ( CFArray without retaining entries ) is used in GIGraph.

```haskell
;; A quick note
GIGraph := ([GIBranch], [GILayer], [GILine], [GINode], SomethingElse)
```

But if we replace CFArray by NSArray, then, I can't find any difference in performance.

Related question
https://stackoverflow.com/questions/16678382/how-does-cfbridgingrelease-act-on-objects-retained-in-a-core-foundation-containe
